test/functional/feature_reindex.py

```diff
@@ -68,12 +72,49 @@ def find_block(b, start):
             'LoadExternalBlockFile: Out of order block',
             'LoadExternalBlockFile: Processing out of order child',
         ]):
-            extra_args = [["-reindex"]]
-            self.start_nodes(extra_args)
+            self.start_node(0, extra_args=["-reindex"])
 
         # All blocks should be accepted and processed.
         assert_equal(self.nodes[0].getblockcount(), 12)
 
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        addr = self.nodes[1].get_deterministic_priv_key().address
+
+        # generate enough blocks to ensure that the -fastprune node fills up the
+        # first blk00000.dat file and starts another block file
+
+        # How big are empty regtest blocks?
+        block_hash = self.generatetoaddress(self.nodes[1], 1, addr)[0]
+        block_size = self.nodes[1].getblock(block_hash)["size"]
+        block_size += 8 # BLOCK_SERIALIZATION_HEADER_SIZE
+
+        # How many blocks do we need to roll over a new .blk file?
+        block_count = self.nodes[1].getblockcount()
+        fastprune_blockfile_size = 0x10000
+        size_needed = fastprune_blockfile_size - (block_size * block_count)
+        blocks_needed = size_needed // block_size
+
+        self.generatetoaddress(self.nodes[1], blocks_needed, addr)
+        self.stop_node(1)
+
+        assert os.path.exists(self.nodes[1].chain_path / 'blocks' / 'blk00000.dat')
+        assert os.path.exists(self.nodes[1].chain_path / 'blocks' / 'blk00001.dat')
```

Comment:
Some assertions are not checking that the PR's changes are working correctly, they're ensuring that the test is setting things up as expected, and sometimes that's not obvious to readers. Sorry I didn't think to mention this earlier, but if you retouch, consider adding a comment before these two lines to make it clear that all we're doing here is verifying the test setup.

Reply:
Ok good feedback. Yeah these asserts are more like sanity checks. Testing the test! Although they also fail early on certain code regressions for example if the `-fastprune` size is changed in the future.